bigquery/benchmarks/bench.go

```diff
@@ -17,69 +17,304 @@
+		}
+	}
+	fmt.Println()
+	return profiles, nil
+}
+
+// printResults prints information about collected query profiles.
+func printResults(queries []*profiledQuery) {
+	for i, prof := range queries {
+		fmt.Printf("%d: (%s:%s)\n", i, prof.GroupName, prof.Name)
+		fmt.Printf("SQL: %s\n", prof.Query.Q)
+		fmt.Printf("MEASUREMENTS\n")
+		for j, timing := range prof.Runs {
+			fmt.Printf("\t\t(%d) %s\n", j, timing.Summary())
+		}
+		fmt.Println()
+	}
+}
+
+// prepareTable ensures a table exists, and optionally creates it if directed
+func prepareTable(ctx context.Context, client *bigquery.Client, table string, create bool) (*bigquery.Table, error) {
+	// Ensure table exists before streaming results, and possibly create it if directed.
+	parts := strings.Split(table, ".")
+	if len(parts) != 3 {
+		return nil, fmt.Errorf("Expected table in p.d.t format, got: %s", table)
+	}
+	tRef := client.DatasetInProject(parts[0], parts[1]).Table(parts[2])
+	// check with backend
+	_, err := tRef.Metadata(ctx)
+	if err != nil {
+		if create {
+			schema, err := bigquery.InferSchema(profiledQuery{})
+			if err != nil {
+				return nil, fmt.Errorf("could not infer schema while creating table: %v", err)
+			}
+			createMeta := &bigquery.TableMetadata{
+				Schema: schema.Relax(),
+				TimePartitioning: &bigquery.TimePartitioning{
+					Type:  bigquery.DayPartitioningType,
+					Field: "event_time",
+				},
+				Clustering: &bigquery.Clustering{
+					Fields: []string{"groupname", "name"},
+				},
+			}
+			if err2 := tRef.Create(ctx, createMeta); err2 != nil {
+				return nil, fmt.Errorf("could not create table: %v", err2)
+			}
+			return tRef, nil
+		}
+		return nil, fmt.Errorf("error while validating table existence: %v", err)
+	}
+	return tRef, nil
+}
+
+func reportResults(ctx context.Context, client *bigquery.Client, table *bigquery.Table, results []*profiledQuery) error {
+
+	inserter := table.Inserter()
+
+	// Set a timeout on our context to bound retries
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+	if err := inserter.Put(ctx, results); err != nil {
+		return fmt.Errorf("reportResults: %v", err)
+	}
+	return nil
+}
+
+func main() {
+	var reruns = flag.Int("reruns", 3, "number of reruns to issue for each query")
+	var queryfile = flag.String("queryfile", "benchmarked-queries.json", "path to file contain queries to be benchmarked.")
+	var projectID = flag.String("projectid", "", "project ID to use for running benchmarks.  Uses PROJECT_ID env if not set.")
+	var reportTable = flag.String("table", "", "table to stream results into, specified in project.dataset.table format")
+	var createTable = flag.Bool("create_table", false, "create result table if it does not exist")
+
+	var tags tags
+	flag.Var(&tags, "tag", "an optional key and value seperated by colon (:) character")
+	flag.Parse()
+
+	// Validate flags.
+	if *reruns <= 0 {
+		log.Fatalf("--reruns should be a positive value")
+	}
+	projID := os.Getenv("PROJECT_ID")
```

Comment:
Should this be `GCLOUD_PROJECT` or `GOOGLE_CLOUD_PROJECT` instead? I've seen those used as default env variables.

Reply:
swapped to GOOGLE_CLOUD_PROJECT